spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -71,15 +74,41 @@ public void close() {
 
 	@Override
 	public boolean isOpen() {
-		return this.delegate != null
-				&& (this.delegate.isOpen() || this.delegate.getClass().getSimpleName().contains("AutorecoveringConnection"));
+		if (this.delegate instanceof AutorecoveringConnection && !this.delegate.isOpen()) {
+			throw new AmqpException("Auto recovery connection is not currently open");
```

Comment:
If a connection is in recovery, we get "connection already closed exception" on channel creation, which is misleading - previously we always returned true for these connections. So I thought this is a more useful error.